mypy/checkexpr.py

```diff
@@ -1694,6 +1694,14 @@ def visit_dict_expr(self, e: DictExpr) -> Type:
 
         Translate it into a call to dict(), with provisions for **expr.
         """
+        if isinstance(self.type_context[-1], TypedDictType):
+            self.check_typeddict_call_with_dict(
+                callee=self.type_context[-1],
+                kwargs=e,
+                context=e
+            )
+            return self.type_context[-1].copy_modified()
```

Comment:
Could you please and a comment here about why you do it like this, and how it works?